File: 14.最长公共前缀.ts
/*
 * @lc app=leetcode.cn id=14 lang=typescript
 *
 * [14] 最长公共前缀
 */

// @lc code=start
function longestCommonPrefix(strs: string[]): string {
    if (strs.length === 0) return "";
    strs.sort();
    let [start, end] = [strs[0], strs[strs.length-1]];
    let i = 0;
    while (i < start.length && i < end.length && start[i] === end[i])
        i++;
    return start.substring(0, i);
};
// @lc code=end

File: 14.最长公共前缀.rs
/*
 * @lc app=leetcode.cn id=14 lang=rust
 *
 * [14] 最长公共前缀
 */

// @lc code=start
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() {
            return "".to_string();
        }

        let mut strs = strs;
        strs.sort();
        let (start, end) = (&strs[0], &strs[strs.len()-1]);
        let mut i = 0;

        while i < start.len() && i < end.len() && 
            start.chars().nth(i) == end.chars().nth(i) {
                i += 1;
        }

        start[0..i].to_string()
    }
}
// @lc code=end

